zenoh-ts/src/encoding.ts

```diff
@@ -0,0 +1,380 @@
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+enum encoding {
+  ZENOH_BYTES = "zenoh/bytes",
+  ZENOH_INT8 = "zenoh/int8",
+  ZENOH_INT16 = "zenoh/int16",
+  ZENOH_INT32 = "zenoh/int32",
+  ZENOH_INT64 = "zenoh/int64",
+  ZENOH_INT128 = "zenoh/int128",
+  ZENOH_UINT8 = "zenoh/uint8",
+  ZENOH_UINT16 = "zenoh/uint16",
+  ZENOH_UINT32 = "zenoh/uint32",
+  ZENOH_UINT64 = "zenoh/uint64",
+  ZENOH_UINT128 = "zenoh/uint128",
+  ZENOH_FLOAT32 = "zenoh/float32",
+  ZENOH_FLOAT64 = "zenoh/float64",
+  ZENOH_BOOL = "zenoh/bool",
+  ZENOH_STRING = "zenoh/string",
+  ZENOH_ERROR = "zenoh/error",
+  APPLICATION_OCTET_STREAM = "application/octet-stream",
+  TEXT_PLAIN = "text/plain",
+  APPLICATION_JSON = "application/json",
+  TEXT_JSON = "text/json",
+  APPLICATION_CDR = "application/cdr",
+  APPLICATION_CBOR = "application/cbor",
+  APPLICATION_YAML = "application/yaml",
+  TEXT_YAML = "text/yaml",
+  TEXT_JSON5 = "text/json5",
+  APPLICATION_PROTOBUF = "application/protobuf",
+  APPLICATION_PYTHON_SERIALIZED_OBJECT = "application/python-serialized-object",
+  APPLICATION_JAVA_SERIALIZED_OBJECT = "application/java-serialized-object",
+  APPLICATION_OPENMETRICS_TEXT = "application/openmetrics-text",
+  IMAGE_PNG = "image/png",
+  IMAGE_JPEG = "image/jpeg",
+  IMAGE_GIF = "image/gif",
+  IMAGE_BMP = "image/bmp",
+  IMAGE_WEBP = "image/webp",
+  APPLICATION_XML = "application/xml",
+  APPLICATION_X_WWW_FORM_URLENCODED = "application/x-www-form-urlencoded",
+  TEXT_HTML = "text/html",
+  TEXT_XML = "text/xml",
+  TEXT_CSS = "text/css",
+  TEXT_JAVASCRIPT = "text/javascript",
+  TEXT_MARKDOWN = "text/markdown",
+  TEXT_CSV = "text/csv",
+  APPLICATION_SQL = "application/sql",
+  APPLICATION_COAP_PAYLOAD = "application/coap-payload",
+  APPLICATION_JSON_PATCH_JSON = "application/json-patch+json",
+  APPLICATION_JSON_SEQ = "application/json-seq",
+  APPLICATION_JSONPATH = "application/jsonpath",
+  APPLICATION_JWT = "application/jwt",
+  APPLICATION_MP4 = "application/mp4",
+  APPLICATION_SOAP_XML = "application/soap+xml",
+  APPLICATION_YANG = "application/yang",
+  AUDIO_AAC = "audio/aac",
+  AUDIO_FLAC = "audio/flac",
+  AUDIO_MP4 = "audio/mp4",
+  AUDIO_OGG = "audio/ogg",
+  AUDIO_VORBIS = "audio/vorbis",
+  VIDEO_H261 = "video/h261",
+  VIDEO_H263 = "video/h263",
+  VIDEO_H264 = "video/h264",
+  VIDEO_H265 = "video/h265",
+  VIDEO_H266 = "video/h266",
+  VIDEO_MP4 = "video/mp4",
+  VIDEO_OGG = "video/ogg",
+  VIDEO_RAW = "video/raw",
+  VIDEO_VP8 = "video/vp8",
+  VIDEO_VP9 = "video/vp9",
+}
+
+export type IntoEncoding = Encoding | String | string;
+
+/** 
+ * Zenoh Encoding Class
+*/
+export class Encoding {
+  private _schema: string;
+
+  private constructor(str_rep: string) {
+    this._schema = str_rep;
+  }
+
+  static into_Encoding(input: IntoEncoding): Encoding {
```

Comment:
```suggestion
  static intoEncoding(input: IntoEncoding): Encoding {
```

In general, I think this library should consistently follow the CamelCase style. It is the style most predominant in the TypeScript documentation and JavaScript browser APIs in general.

There are other sites where this comment applies.

Reply:
Agreed, opened a PR with this. 
https://github.com/eclipse-zenoh/zenoh-ts/pull/5

Reply:
Implemented in https://github.com/eclipse-zenoh/zenoh-ts/pull/5